witness/src/Dscp/Snowdrop/Expanders.hs

```diff
@@ -170,16 +227,45 @@ seqExpandersBalanceTx =
             pure (AccountId txOutAddr ==> ch)
 
         let txId = toGTxId $ GMoneyTx twTx
-
-        accList <- forM (inAddr : map txOutAddr outOther) $ \addr ->
-            queryOne (TxsOf addr) >>= return . \case
-                Nothing -> [TxsOf addr ==> New (LastTx txId)]
+        miscChanges <- forM (inAddr : map txOutAddr outOther) $ \txAddr ->
+            queryOne (TxsOf txAddr) >>= return . \case
+                Nothing -> [TxsOf txAddr ==> New (LastTx txId)]
                 Just (LastTx prevTxId) ->
-                    [ TxsOf  addr      ==> Upd (LastTx txId)
-                    , TxHead addr txId ==> New (TxNext prevTxId)
+                    [ TxsOf  txAddr      ==> Upd (LastTx txId)
+                    , TxHead txAddr txId ==> New (TxNext prevTxId)
                     ]
 
-        pure $ mkDiffCS $ Map.fromList $ inp : outs ++ concat accList
+        -- Money flow related changes (w/o fees)
+        let changesList = inp : outs ++ concat miscChanges
+
+        -- Checking for duplicates
+        let hasDuplicates l = length l /= length (ordNub l)
```

Comment:
I would appreciate if you move it to `Dscp.Util`, it's often useful

Reply:
It seemed that we had it in srk-utils at some point :thinking: